pkg/engine/virtualmachinescalesets.go

```diff
@@ -385,6 +389,10 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 	k8sConfig := orchProfile.KubernetesConfig
 	linuxProfile := cs.Properties.LinuxProfile
 
+	if k8sConfig != nil && k8sConfig.UseManagedIdentity && k8sConfig.UserAssignedID != "" {
```

Comment:
there is a util method UserAssignedIDEnabled() in pkg/api/types.go, should we reuse that?

Reply:
yeah, it makes sense, updated.